pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerGauge.java

```diff
@@ -153,7 +153,11 @@ public enum ServerGauge implements AbstractMetrics.Gauge {
 
   // ThrottleOnCriticalHeapUsageExecutor metrics
   THROTTLE_EXECUTOR_QUEUE_SIZE("count", true,
-      "Current number of tasks in the throttle executor queue");
+      "Current number of tasks in the throttle executor queue"),
+
+  // Multi-stage executor thread usage metrics
+  MSE_THREAD_USAGE_MAX("threads", true, "Maximum allowed threads for multi-stage executor"),
```

Comment:
Since this is fixed, I feel it is not worth adding given the overhead of processing gauge. We can log it when HardLimitExecutor is initialized

Reply:
we would like to keep this as without it you can't compute the % of limit used. Ideally we can setup alerts/detectors when users are approaching their limits and action it before they notice. If this metric is not available, we wont be able to

do you antipciate processing overhead even if its just set once during constructor?